math/mathcore/inc/Math/DistSampler.h

```diff
@@ -190,27 +215,34 @@ class DistSampler {
       return true;
    }
    /**
-      sample a set of bins given a vector of probabilities
+      Sample a set of bins given a vector of probabilities
       Typically multinomial statistics will be used and the sum of the probabilities
       will be equal to the total number of events to be generated
       For sampling the bins indipendently, SampleBin should be used
     */
    virtual bool SampleBins(unsigned int n, const double * prob, double * values, double * errors  = 0)  {
-      std::copy(prob,prob+n, values);
+      std::copy(prob,prob+n, values); // default impl returns prob values (Asimov data)
       if (errors) std::fill(errors,errors+n,0);
       return true;
    }
 
 
    /**
-      generate a un-binned data sets (fill the given data set)
-      if dataset has already data append to it
+      Generate a un-binned data sets by fill the given data set.
```

Comment:
@lmoneta I think there should be "Generate an un-binned data set by filling the given data set."